include/fc/io/raw.hpp

```diff
@@ -193,10 +193,10 @@ namespace fc {
       uint64_t v = 0; char b = 0; uint8_t by = 0;
       do {
           s.get(b);
-          v |= uint32_t(uint8_t(b) & 0x7f) << by;
+          v |= uint64_t(uint8_t(b) & 0x7f) << by;
```

Comment:
How about checking `by` before the bit shift? For example, if `b` is `0x7f` and `by` is `7*9=63`, it will overflow, so result of the bit shift will be `1`.

BTW I suspected that https://github.com/bitshares/bitshares-core/issues/1256 was caused by something like this, but I guess it won't be the case due to limit of maximum message size.

Reply:
https://en.cppreference.com/w/cpp/language/operator_arithmetic
> For unsigned and positive a, the value of a << b is the value of a * 2b
, reduced modulo maximum value of the return type plus 1 (that is, bitwise left shift is performed and the bits that get shifted out of the destination type are discarded). 

So for uint64_t, (0x7f << 63) == (1 << 63).

Reply:
Hm, that means this change still doesn't enforce a canonical representation. Not sure if that was the intention.

Reply:
So `(0x7e << 63) == 0` which doesn't seem right to me (talking about our code but not the `<<` operator itself).

Reply:
Still, how about throwing an exception when the input is not valid, e.g. too large?